### PR TITLE
Add methods for setting and getting variants of a move

### PR DESCRIPTION
**Pull request by stanleydesu**

### CODE REVIEW

File: src/generic_cube.rs [hunk 1]
```diff
@@ -185,13 +185,56 @@ pub enum Move {
...
+    }
+}
 
 /// A move variation that must be applied to the ```Move``` struct.
 #[allow(dead_code)]
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 pub enum MoveVariant {
     /// A 90 degree clockwise turn.
-    Standard,
+    Standard = 1,
```
Comment (V-Wong) [inline src/generic_cube.rs:237]: This is a good idea that should simplify a lot of the code.

File: src/generic_cube.rs [hunk 2]
```diff
@@ -185,13 +185,56 @@ pub enum Move {
...
+            | Move::Lw(_, v)
+            | Move::Fw(_, v)
+            | Move::Rw(_, v)
+            | Move::Bw(_, v)
+            | Move::Dw(_, v) => *v,
+        }
+    }
+
+    /// Returns the Move with the given MoveVariant.
+    pub fn with_variant(&self, variant: MoveVariant) -> Move {
```
Comment (V-Wong) [inline src/generic_cube.rs:211]: This is acceptable, but there might be a better way to do this. I'll come back to this with comments later.

File: src/generic_cube.rs [hunk 3]
```diff
@@ -185,13 +185,56 @@ pub enum Move {
     Z(MoveVariant),
 }
 
+impl Move {
+    /// Extracts the MoveVariant of a Move.
+    pub fn get_variant(&self) -> MoveVariant {
```
Comment (V-Wong) [inline src/generic_cube.rs:190]: This is acceptable, but there might be a better way to do this. I'll come back to this with comments later.

Comment (stanleydesu) [reply src/generic_cube.rs:190]: Other than restructuring the Move enum, there isn't.

File: src/generic_cube.rs (hunk 2 above)
Comment (stanleydesu) [reply src/generic_cube.rs:211]: Other than restructuring the Move enum, there isn't.

**Pull request merged**